zkevm-circuits/src/evm_circuit/execution/error_oog_call.rs

```diff
@@ -78,16 +80,24 @@ impl<F: Field> ExecutionGadget<F> for ErrorOOGCallGadget<F> {
 
         // Check if the amount of gas available is less than the amount of gas required
         let insufficient_gas = LtGadget::construct(cb, cb.curr.state.gas_left.expr(), gas_cost);
+
         cb.require_equal(
-            "gas left is less than gas required",
-            insufficient_gas.expr(),
+            "Either Memory address is overflow or gas left is less than cost",
+            or::expr([
+                call_gadget.cd_address.overflow(),
+                call_gadget.rd_address.overflow(),
+                insufficient_gas.expr(),
+            ]),
             1.expr(),
         );
 
         // Both CALL and CALLCODE opcodes have an extra stack pop `value` relative to
         // DELEGATECALL and STATICCALL.
-        let common_error_gadget =
-            CommonErrorGadget::construct(cb, opcode.expr(), cb.rw_counter_offset());
+        let common_error_gadget = CommonErrorGadget::construct(
+            cb,
+            opcode.expr(),
+            11.expr() + is_call.expr() + is_callcode.expr(),
```

Comment:
I noticed we fall back to manual offset counting here. Is it because `cb.rw_counter_offset()` doesn't work here?

Reply:
no, it works,  but I feel `cb.rw_counter_offset()` is indirect way for calculating the rw offset.  we can not know how the `rwc` change from the code directly, it is always right.  I prefer old way of it.

Reply:
issue #1599 for discussion .